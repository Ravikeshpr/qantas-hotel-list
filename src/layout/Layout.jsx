import React from "react";
import { MainContainer } from "./Layout.style";
import Header from "./Header";

function Layout({ children }) {
    return (
        <>
            <MainContainer>
                <Header />
                <main data-testid="main-content">{children}</main>
            </MainContainer>
        </>
    );
}

export default Layout;
