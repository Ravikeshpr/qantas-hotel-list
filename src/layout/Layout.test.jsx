import React from "react";
import { render, screen } from "@testing-library/react";
import Layout from "./Layout";
import { describe, it, expect } from "vitest";
import "@testing-library/jest-dom/vitest";

describe("Layout", () => {
    it("Layout childern should be wrapped in main element", () => {
        render(<Layout />);
        expect(screen.getByTestId("main-content")).toBeInTheDocument();
    });

    it("Header with logo should be part of Layout component", () => {
        render(<Layout />);
        expect(screen.getByAltText("Qantas logo")).toBeInTheDocument();
    });

    it("Layout component should match the snapshot", () => {
        const { container } = render(<Layout />);
        expect(container).toMatchSnapshot();
    });
});
