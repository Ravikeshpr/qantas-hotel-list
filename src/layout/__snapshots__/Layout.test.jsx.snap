// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout > Layout component should match the snapshot 1`] = `
<div>
  <div
    class="sc-VHjGu jhXzHT"
  >
    <header
      class="sc-gnOvAp bDeckY"
      role="banner"
    >
      <img
        alt="Qantas logo"
        class="sc-dQkurY kkMorT"
        role="logo"
        src="/src/assets/qantas-logo.png"
        tabindex="0"
      />
    </header>
    <main
      data-testid="main-content"
    />
  </div>
</div>
`;
